src/function/arithmetic/ceil.js

```diff
@@ -122,6 +128,24 @@ export const createCeil = /* #__PURE__ */ factory(name, dependencies, ({ typed,
       return x.ceil(n.toNumber())
     },
 
+    'Unit, number, Unit': typed.referToSelf(self => function (x, n, unit) {
```

Comment:
I think the code can be simplified from 6 new signatures to 3:

```js
'Unit, number | BigNumber, Unit': ...,
'Array | Matrix, number | BigNumber, Unit': ...,
'Array | Matrix | Unit, Unit': ...
```

And the same for functions `fix` and `floor`. Can you give that a try?